src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -344,133 +344,304 @@ public class SqlMutationEngine : IMutationEngine
                     return new NoContentResult();
                 }
             }
-            else if (context.OperationType is EntityActionOperation.Upsert || context.OperationType is EntityActionOperation.UpsertIncremental)
+            else
             {
-                DbResultSet? upsertOperationResult = null;
+                string roleName = GetHttpContext().Request.Headers[AuthorizationResolver.CLIENT_ROLE_HEADER];
+                bool isReadPermissionConfiguredForRole = _authorizationResolver.AreRoleAndOperationDefinedForEntity(context.EntityName, roleName, EntityActionOperation.Read);
+                bool isDatabasePolicyDefinedForReadAction = false;
 
-                try
+                if (isReadPermissionConfiguredForRole)
                 {
-                    // Creating an implicit transaction
-                    using (TransactionScope transactionScope = ConstructTransactionScopeBasedOnDbType())
-                    {
-                        upsertOperationResult = await PerformUpsertOperation(
-                                                            parameters,
-                                                            context);
-                        transactionScope.Complete();
-                    }
+                    isDatabasePolicyDefinedForReadAction = _authorizationResolver.IsDBPolicyDefinedForRoleAndAction(context.EntityName, roleName, EntityActionOperation.Read);
```

Comment:
`!string.IsNullOrWhiteSpace(_authorizationResolver.GetDBPolicyForRequest(entityName, roleName, operation));`

why not do something like this? with the benefit of:
1. AuthorizationResolver doesn't need to know implementation details of what you're doing here.
2. Don't need to modify the `IauthorizationResolver` contract.

Reply:
Thanks for the suggestion!

`GetDBPolicyForRequest` is a private method in `AuthorizationResolver` which is an implementation of `IAuthorizationResolver`.  To be able to use this in `SqlMutationEngine`, this method has to be made `public`; which would lead to changes in the interface contract. Hence, I decided to add a public wrapper over the private method.  

Both the ways ( adding a public wrapper  (or)  switching the method from private to public ) would force all the implementations of this interface to provide an implementation for the method. 

Considering that it can be deterimed whether a database policy is defined for a (role , operation) from by  retrieving the policy defined, I've decided to have the `GetDBPolicyForRequest` (flip it to a public method) in the interface.